crates/rpc/rpc/src/debug.rs

```diff
@@ -64,16 +70,28 @@ impl<Provider, Eth> DebugApi<Provider, Eth> {
 
 impl<Provider, Eth> DebugApi<Provider, Eth>
 where
-    Provider: BlockReaderIdExt + HeaderProvider + ChainSpecProvider + 'static,
-    Eth: EthTransactions + LoadState + SpawnBlocking + LoadPendingBlock + 'static,
+    Provider: BlockReaderIdExt
+        + HeaderProvider
+        + ChainSpecProvider
+        + StateProviderFactory
+        + EvmEnvProvider
+        + 'static,
+    Eth: LoadState
+        + LoadPendingBlock
+        + LoadTransaction
+        + LoadBlock
+        + Trace
+        + Call
+        + SpawnBlocking
```

Comment:
okay, not this is coming together

can we add a few more docs that explain why we have multiple traits, basically so that we can reuse all of these API impls with any `Eth` that implements these features